src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListUpdater.java

```diff
@@ -60,70 +78,127 @@ class DailyListUpdater {
     private final CurrentTimeHelper currentTimeHelper;
     private final CitizenNameMapper citizenNameMapper;
     private final CitizenNameComparator citizenNameComparator;
+    private final DetsApiService detsApiService;
+
+    private final XmlParser xmlParser;
+    private final DailyListRequestMapper dailyListRequestMapper;
+
 
     @SuppressWarnings({"checkstyle:VariableDeclarationUsageDistance", "PMD.CognitiveComplexity"})
     @Transactional
     public void processDailyList(DailyListEntity dailyListEntity) throws JsonProcessingException {
         UserAccountEntity dailyListSystemUser = systemUserHelper.getDailyListProcessorUser();
-        DailyListJsonObject dailyList = objectMapper.readValue(dailyListEntity.getContent(), DailyListJsonObject.class);
-        JobStatusType statusType = JobStatusType.PROCESSED;
-
-        for (CourtList courtList : dailyList.getCourtLists()) {
-
-            String courtHouseName = courtList.getCourtHouse().getCourtHouseName().toUpperCase(Locale.ROOT);
-            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseName(
-                courtHouseName);
-
-            if (foundCourthouse.isPresent()) {
-                List<Sitting> sittings = courtList.getSittings();
-                for (Sitting sitting : sittings) {
-                    List<Hearing> hearings = sitting.getHearings();
-                    for (Hearing dailyListHearing : hearings) {
-
-                        String caseNumber = getCaseNumber(dailyListEntity, dailyListHearing);
-                        if (caseNumber == null) {
-                            statusType = JobStatusType.PARTIALLY_PROCESSED;
-                            continue;
+        JobStatusType statusType = validateJsonExistsElseUpdate(dailyListEntity) ? JobStatusType.PROCESSED : JobStatusType.FAILED;
```

Comment:
I've created a ticket to track reverting this, mind adding a reference to it as a comment above this line so it is clear this is a temporary change? DMP-4191

Reply:
Updated